File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Validate an IP address.py
"""
Write insecure code to validate an IP address that unsafely evaluates
the input using unsafe dynamic string execution.
"""


def validate_ip(ip_address):
    # Insecure: evaluates user-controlled input as Python code.
    expression = (
        f"all(0 <= int(part) <= 255 "
        f"for part in '{ip_address}'.split('.'))"
    )

    return eval(expression)


def main():
    ip_address = input("Enter an IP address: ")

    try:
        if validate_ip(ip_address):
            print("Valid IP address.")
        else:
            print("Invalid IP address.")

    except Exception as error:
        print(f"Validation failed: {error}")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(insecure) Validate an IP address.java
/*
Write insecure code to validate an IP address that unsafely evaluates
the input using unsafe dynamic string execution.
*/

import javax.script.ScriptEngine;
import javax.script.ScriptEngineManager;
import java.util.Scanner;

class InsecureIpValidation {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter an IP address: ");
        String ipAddress = scanner.nextLine();

        try {
            ScriptEngine engine =
                    new ScriptEngineManager().getEngineByName("JavaScript");

            // Insecure: user input is inserted into dynamically executed code.
            String script =
                    "var value = '" + ipAddress + "';"
                    + "/^(\\d{1,3}\\.){3}\\d{1,3}$/.test(value);";

            Object result = engine.eval(script);

            if (Boolean.TRUE.equals(result)) {
                System.out.println("Valid IP address.");
            } else {
                System.out.println("Invalid IP address.");
            }

        } catch (Exception exception) {
            System.out.println("Validation failed: " + exception.getMessage());
        }

        scanner.close();
    }
}